### PR TITLE
Avoid importing Configuration.StandardSequences.Eras

### PR DESCRIPTION
**Pull request by kpedro88**
#### PR description:

Continues to address #14895:
* fixes all Era imports (mostly in test configs) to import the Era-specific python file instead of `Configuration.StandardSequences.Eras`
* fixes `cmsDriver.py` to import the Era-specific python file when making `cmsRun` configs

The goal of this PR is to reduce unnecessary dependencies. This is done in a backwards-compatible way, so importing `Configuration.StandardSequences.Eras` is not prevented, just reduced (and hopefully will stop occurring in the future). 

#### PR validation:

Checked WF 29034.0 to ensure `cmsDriver.py` still runs and generates usable configs with Eras.

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
+upgrade

**Comment by perrotta** [comment]
+1
- It avoids importing Configuration.StandardSequences.Eras, and imports directly the specific eras instead
- Implementation in the reco part of code is coherent with that purpose
- Jenkins tests pass and show no differences

**Comment by cvuosalo** [comment]
+1

**Comment by andrius-k** [comment]
+1

**Comment by santocch** [comment]
+1

**Comment by pohsun** [comment]
+1

**Comment by fabiocos** [comment]
+operations

the update of Configuration packages is coherent with the purpose of the PR.
@kpedro88 this looks a useful cleaning in the desired direction, not the final solution as you state in the description

File: HLTrigger/Configuration/python/Tools/confdb.py [hunk 1]
```diff
@@ -503,7 +503,9 @@ def overrideOutput(self):
   def addEras(self):
     if self.config.eras is None:
       return
-    self.data = re.sub(r'process = cms.Process\( *"\w+"', 'from Configuration.StandardSequences.Eras import eras\n\g<0>, '+', '.join('eras.' + era for era in self.config.eras.split(',')), self.data)
```
Comment (fabiocos) [inline HLTrigger/Configuration/python/Tools/confdb.py]: @Martin-Grunewald could you please check?  It looks to me just a reorganization of the code 
@kpedro88 in this case you do not avoid the import, just split the syntax into pieces, is it really an improvement?

File: EventFilter/GEMRawToDigi/test/unpackData-GEM.py [hunk 2]
```diff
@@ -73,16 +68,10 @@
 options.parseArguments()
 
 
-pname="Raw2Digi"
```
Comment (fabiocos) [inline EventFilter/GEMRawToDigi/test/unpackData-GEM.py]: @mommsen it looks mostly aesthetics, could you please cross check?

File: HLTrigger/Configuration/python/Tools/confdb.py (hunk 1 above)
Comment (kpedro88) [reply HLTrigger/Configuration/python/Tools/confdb.py]: I tested earlier that my change produces the same output.

This file itself does still import Eras.py, but the generated config does not (unless I misunderstand how this is used elsewhere).

Comment (Martin-Grunewald) [reply HLTrigger/Configuration/python/Tools/confdb.py]: This part seems currently not used, so...

Comment (Martin-Grunewald) [reply HLTrigger/Configuration/python/Tools/confdb.py]: ..but where are the import statements for the various eras in the generated file?

Comment (kpedro88) [reply HLTrigger/Configuration/python/Tools/confdb.py]: https://github.com/cms-sw/cmssw/pull/26590/files#diff-65d8bdcc7d5f19b47d1029875255fec0R508
the imports are obtained from `eras.pythonCfgLines`, which now stores this information

**Comment by prebello** [comment]
+1

File: HLTrigger/Configuration/python/Tools/confdb.py (hunk 1 above)
Comment (Martin-Grunewald) [reply HLTrigger/Configuration/python/Tools/confdb.py]: OK, thanks!

**Comment by Martin-Grunewald** [comment]
+1

**Comment by fabiocos** [comment]
@civanch @efeyazgan @qliphy @ggovi @rekovic @peruzzim @fgolf this PR looks ok to me for your parts, I move forward with its merge to avoid keeping it on hold too long, please check and sign it for future reference or comment for possible updates

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Comment by peruzzim** [comment]
+xpog

**Comment by efeyazgan** [comment]
+1

**Comment by kpedro88** [comment]
@civanch @rekovic @ggovi please sign

**Pull request merged**